<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Mengyin Lu" />

<meta name="date" content="2018-08-07" />

<title>Use FASH to analyze the stem cell gene expression data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">general_ash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Use FASH to analyze the stem cell gene expression data</h1>
<h4 class="author"><em>Mengyin Lu</em></h4>
<h4 class="date"><em>2018-08-07</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-08-07</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180705)</code> </summary></p>
<p>The command <code>set.seed(20180705)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/mengyin/general_ash/tree/7836ca3351a4995f184bd7b25a0f64537658dc29" target="_blank">7836ca3</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/stem_cache/

Unstaged changes:
    Modified:   analysis/stem.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/mengyin/general_ash/7836ca3351a4995f184bd7b25a0f64537658dc29/docs/stem.html" target="_blank">7836ca3</a>
</td>
<td style="text-align:left;">
mengyin
</td>
<td style="text-align:left;">
2018-08-07
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/mengyin/general_ash/blob/92644611d4a57984a9dd47c97ce9d1ae75d04440/analysis/stem.Rmd" target="_blank">9264461</a>
</td>
<td style="text-align:left;">
mengyin
</td>
<td style="text-align:left;">
2018-08-07
</td>
<td style="text-align:left;">
Add stem data analysis using FASH
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="data-description" class="section level2">
<h2>Data description</h2>
<p>We have gene expression (microarray) data from <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005793">Burrows et al (2016), “Genetic Variation, Not Cell Type of Origin, Underlies the Majority of Identifiable Regulatory Differences in iPSCs”</a>.</p>
<p>There are four individuals - Ind1, Ind2, Ind3, Ind4. Each individual has four samples types - Fibroblast, LCL, F-iPSC, L-iPSC. There are three replicates of the L-iPSCs A-C.</p>
<p>L-iPSC referes to iPSCs derived from LCLs, F-iPSC refers to iPSCs derived from Fibroblasts.</p>
<p>They were interested in the proportion of variation explained by cell type of origin vs the proportion of variation explained by individual in the iPSCs. They performed a linear mixed model with a fixed effect for cell type of origin (ie L-iPSC vs F-iPSC) and a random effect for individual. This model did no use the LCLs or the Fibroblasts from these individuals.</p>
</div>
<div id="data-analysis" class="section level2">
<h2>Data analysis</h2>
<p>We use the naive F-test (ANOVA) and FASH to analyze the proportion of variance explained by cell type or individual. Specifically, we assume that gene expression <span class="math inline">\(y_{gij}\)</span> comes from the following model: <span class="math display">\[y_{gij} = \mu_g+\beta_{g,c(i)}+\gamma_{g,d(j)}+e_{gij},\]</span> where <span class="math inline">\(g,i,j\)</span> are the indices for gene, individual and cell type respectively. <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> are random effects for individuals and celltypes respectively. Suppose <span class="math inline">\(\beta_{g,c(i)}\sim N(0,v_{g}^{(ind)}\)</span>, <span class="math inline">\(\gamma_{g,c(i)}\sim N(0,v_{g}^(ct)\)</span> and <span class="math inline">\(e_{gij}\sim N(0, v_{g}^{(err)})\)</span>, we are interested in estimating the “PVE”s defined as follows: <span class="math display">\[PVE_g^{(ind)}:=\frac{v_{g}^{(ind)}}{v_{g}^{(ind)}+v_{g}^{(ct)}+v_{g}^{(err)}},\]</span> <span class="math display">\[PVE_g^{(ct)}:=\frac{v_{g}^{(ct)}}{v_{g}^{(ind)}+v_{g}^{(ct)}+v_{g}^{(err)}}.\]</span></p>
<p>Below are the R functions for our data analysis:</p>
<pre class="r"><code>library(limma)
library(qvalue)
library(ashr)

# naive F test for a single gene
get_fstat = function(y,condition1,condition2){
  fit = lm(y~condition1+condition2)
  fstat1 = anova(fit)$F[1]
  fstat2 = anova(fit)$F[2]
  MST1 = anova(fit)$M[1]
  MST2 = anova(fit)$M[2]
  MSE = anova(fit)$M[3]
  pval1 = unlist(anova(fit)[&quot;Pr(&gt;F)&quot;])[1]
  pval2 = unlist(anova(fit)[&quot;Pr(&gt;F)&quot;])[2]
  res = c(fstat1,fstat2,MST1,MST2,MSE,pval1,pval2)
  names(res) = c(&quot;fstat1&quot;,&quot;fstat2&quot;,&quot;MST1&quot;,&quot;MST2&quot;,&quot;MSE&quot;,&quot;pval1&quot;,&quot;pval2&quot;)
  return(res)
}

# Variance ratio = var_trt/var_err
# Variance ratio from the raw MSTs &amp; MSEs
vr_ftest = function(nsamp,MST,MSE){
  sigma.c2.hat = pmax(0,(MST-MSE)/nsamp)
  sigma.e2.hat = MSE
  return(sigma.c2.hat/sigma.e2.hat)
}

# Variance ratio = var_trt/var_err
# logalpha = (1+nsamp*var_trt/var_err)
# So: VR = (exp(logalpha)-1)/nsamp
vr_plugin = function(nsamp,fashobj){
  (exp(fashobj$result$PosteriorMean)-1)/nsamp
}

# main function for computing the PVEs
pve_analysis = function(data){
  celltype1 = factor(rep(c(&quot;F&quot;,&quot;L&quot;), each=4))
  individual1 = factor(rep(1:4, 2))
  design1 = model.matrix(~celltype1+individual1)
  
  # naive F tests
  ftests = apply(data,1,get_fstat,condition1=celltype1,condition2=individual1)
  vr.ct.ftest = vr_ftest(4, ftests[&quot;MST1&quot;,],ftests[&quot;MSE&quot;,])
  vr.ind.ftest = vr_ftest(2, ftests[&quot;MST2&quot;,],ftests[&quot;MSE&quot;,])
  
  # fash
  lim = lmFit(data, design=design1)
  lim.ct = eBayes(lim[,2])
  Fstats.ct = lim.ct$F # shrunk F stats
  Fpval.ct = lim.ct$F.p.value
  fash.ct = ash(log(Fstats.ct),1,
                lik=lik_logF(df1=1,df2=lim.ct$df.total[1]),mixcompdist=&quot;+uniform&quot;)
  vr.ct.fash = vr_plugin(4, fash.ct)

  lim.ind = eBayes(lim[,3:5])
  Fstats.ind = lim.ind$F # shrunk F stats
  Fpval.ind = lim.ind$F.p.value
  fash.ind = ash(log(Fstats.ind),1,
                 lik=lik_logF(df1=3,df2=lim.ind$df.total[1]),mixcompdist=&quot;+uniform&quot;)
  vr.ind.fash = vr_plugin(2, fash.ind)
  
  # compute PVEs from variance ratios
  pve.ct.fash = vr.ct.fash/(vr.ct.fash+vr.ind.fash+1)
  pve.ct.ftest = vr.ct.ftest/(vr.ct.ftest+vr.ind.ftest+1)
  pve.ind.fash = vr.ind.fash/(vr.ct.fash+vr.ind.fash+1)
  pve.ind.ftest = vr.ind.ftest/(vr.ct.ftest+vr.ind.ftest+1)
  
  # boxplot
  # boxplot(pve.ct.fash, pve.ct.ftest, pve.ind.fash, pve.ind.ftest,
  #       names=c(&quot;ct.fash&quot;,&quot;ct.ftest&quot;,&quot;ind.fash&quot;,&quot;ind.ftest&quot;),
  #       main=&quot;PVE of celltypes (ct) and individuals (ind)&quot;)

  return(list(pve.ind.ftest=pve.ind.ftest, pve.ind.fash=pve.ind.fash,
              pve.ct.ftest=pve.ct.ftest, pve.ct.fash=pve.ct.fash))
}</code></pre>
<p>Note that this dataset has unbalanced design (three L-iPSC replicates but just one F-iPSC sample for each individual). However it is not straightforward to use FASH on unbalanced dataset for PVE analysis. Hence we choose an ad-hoc way: each time we simply use one of the three L-iPSC replicates to form a balanced dataset, and compare the results of three trials. Fortunately the three trials give very similar results.</p>
<pre class="r"><code># load data
geneexp = read.delim(&quot;data/OriginGeneExpression_Normalized.txt&quot;)

# just use L-iPSC replicate A
geneexp.A = geneexp[,c(&quot;Ind1.F.iPSC&quot;,&quot;Ind2.F.iPSC&quot;,&quot;Ind3.F.iPSC&quot;,&quot;Ind4.F.iPSC&quot;,
                     &quot;Ind1.L.iPSC.A&quot;,&quot;Ind2.L.iPSC.A&quot;,&quot;Ind3.L.iPSC.A&quot;,&quot;Ind4.L.iPSC.A&quot;)]

# just use L-iPSC replicate B
geneexp.B = geneexp[,c(&quot;Ind1.F.iPSC&quot;,&quot;Ind2.F.iPSC&quot;,&quot;Ind3.F.iPSC&quot;,&quot;Ind4.F.iPSC&quot;,
                     &quot;Ind1.L.iPSC.B&quot;,&quot;Ind2.L.iPSC.B&quot;,&quot;Ind3.L.iPSC.B&quot;,&quot;Ind4.L.iPSC.B&quot;)]

# just use L-iPSC replicate C
geneexp.C = geneexp[,c(&quot;Ind1.F.iPSC&quot;,&quot;Ind2.F.iPSC&quot;,&quot;Ind3.F.iPSC&quot;,&quot;Ind4.F.iPSC&quot;,
                     &quot;Ind1.L.iPSC.C&quot;,&quot;Ind2.L.iPSC.C&quot;,&quot;Ind3.L.iPSC.C&quot;,&quot;Ind4.L.iPSC.C&quot;)]

A = pve_analysis(geneexp.A) 
B = pve_analysis(geneexp.B) 
C = pve_analysis(geneexp.C) 

# show distribution of pve estimates by fash or F-test
boxplot(A$pve.ct.fash, A$pve.ct.ftest, A$pve.ind.fash, A$pve.ind.ftest,
        names=c(&quot;CT(fash)&quot;,&quot;CT(F-test)&quot;,&quot;IND(fash)&quot;,&quot;IND(F-test)&quot;), 
        ylab=&quot;Estimated PVE&quot;,
        main=&quot;PVE estimates (with L-iPSC A)&quot;)</code></pre>
<p><img src="figure/stem.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(B$pve.ct.fash, B$pve.ct.ftest, B$pve.ind.fash, B$pve.ind.ftest,
        names=c(&quot;CT(fash)&quot;,&quot;CT(F-test)&quot;,&quot;IND(fash)&quot;,&quot;IND(F-test)&quot;), 
        ylab=&quot;Estimated PVE&quot;,
        main=&quot;PVE estimates (with L-iPSC B)&quot;)</code></pre>
<p><img src="figure/stem.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(C$pve.ct.fash, B$pve.ct.ftest, C$pve.ind.fash, C$pve.ind.ftest,
        names=c(&quot;CT(fash)&quot;,&quot;CT(F-test)&quot;,&quot;IND(fash)&quot;,&quot;IND(F-test)&quot;), 
        ylab=&quot;Estimated PVE&quot;,
        main=&quot;PVE estimates (with L-iPSC C)&quot;)</code></pre>
<p><img src="figure/stem.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Burrows et al (2016) simply uses the limma DE pipeline and claims that only one gene is classified as DE between L-iPSC and F-iPSC. They also use simple ANOVA <span class="math inline">\(R^2\)</span> to record the variance explained by celltypes or individuals (note that <span class="math inline">\(R^2\)</span> is different from our defined “PVE” here) and conclude that “individual genetic background captures a much larger proportion of gene regulatory variation than cell type of origin”.</p>
<p>We see that the FASH results are generally consistent with their results, that almost all genes have almost zero celltype PVEs. Compared to the raw F-test PVE estimates (which are pretty noisy), I think FASH gives much more intuitive results.</p>
<p>According to FASH results, both datasets with L-iPSC replicate A or C suggest that gene “ENSG00000198825” is differentially expressed between L-iPSC and F-iPSC (FDR=0.05), but the dataset with L-iPSC replicate B does not any give significant DE genes (FDR=0.05). We check this gene:</p>
<pre class="r"><code>geneexp[which.max(A$pve.ct.fash),]</code></pre>
<pre><code>                Ind1.F.iPSC Ind2.L.iPSC.C Ind3.L.iPSC.B Ind1.LCL
ENSG00000198825     7.97047      6.902654      6.945877 7.151865
                Ind4.L.iPSC.C Ind2.L.iPSC.B Ind4.L.iPSC.A Ind1.Fibroblast
ENSG00000198825      6.920788      8.395853      6.922644         7.12799
                Ind1.L.iPSC.A Ind4.Fibroblast Ind2.LCL Ind3.F.iPSC
ENSG00000198825      6.997868        7.122821 7.131635    8.029191
                Ind2.Fibroblast Ind3.LCL Ind2.L.iPSC.A Ind4.LCL
ENSG00000198825        7.067461 7.097692      6.900132 7.257512
                Ind3.Fibroblast Ind1.L.iPSC.C Ind2.F.iPSC Ind3.L.iPSC.A
ENSG00000198825        7.115315      7.044821    7.757269      7.045194
                Ind1.L.iPSC.B Ind4.L.iPSC.B Ind3.L.iPSC.C Ind4.F.iPSC
ENSG00000198825      6.969298      6.974893      7.049418    7.896776</code></pre>
<p>For this gene, individual 2 L-iPSC replicate B’s gene expression level is much higher (&gt;1.4) than that of replicates A and C. Compared to the typical gene expression variation within L-iPSC replicates, this seems somewhat suspicious…?</p>
<pre class="r"><code>hist(apply(geneexp[,c(&quot;Ind2.L.iPSC.A&quot;,&quot;Ind2.L.iPSC.B&quot;,&quot;Ind2.L.iPSC.C&quot;)],1,sd), 30,
     xlab=&quot;sd of replicates A-C for ind2&quot;,
     main=&quot;Histogram of sd&quot;)</code></pre>
<p><img src="figure/stem.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/mengyin/general_ash/blob/7836ca3351a4995f184bd7b25a0f64537658dc29/docs/figure/stem.Rmd/unnamed-chunk-4-1.png" target="_blank">7836ca3</a>
</td>
<td style="text-align:left;">
mengyin
</td>
<td style="text-align:left;">
2018-08-07
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.3

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ashr_2.2-8    qvalue_2.12.0 limma_3.36.2 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.18      compiler_3.5.0    pillar_1.3.0     
 [4] git2r_0.23.0      plyr_1.8.4        workflowr_1.1.1  
 [7] bindr_0.1.1       iterators_1.0.10  R.methodsS3_1.7.1
[10] R.utils_2.6.0     tools_3.5.0       digest_0.6.15    
[13] lattice_0.20-35   evaluate_0.11     tibble_1.4.2     
[16] gtable_0.2.0      pkgconfig_2.0.1   rlang_0.2.1      
[19] foreach_1.4.4     Matrix_1.2-14     rstudioapi_0.7   
[22] parallel_3.5.0    yaml_2.2.0        bindrcpp_0.2.2   
[25] stringr_1.3.1     dplyr_0.7.6       knitr_1.20       
[28] tidyselect_0.2.4  rprojroot_1.3-2   grid_3.5.0       
[31] glue_1.3.0        R6_2.2.2          rmarkdown_1.10   
[34] reshape2_1.4.3    purrr_0.2.5       ggplot2_3.0.0    
[37] magrittr_1.5      whisker_0.3-2     MASS_7.3-50      
[40] codetools_0.2-15  splines_3.5.0     backports_1.1.2  
[43] scales_0.5.0      htmltools_0.3.6   assertthat_0.2.0 
[46] colorspace_1.3-2  stringi_1.2.4     pscl_1.5.2       
[49] doParallel_1.0.11 lazyeval_0.2.1    munsell_0.5.0    
[52] truncnorm_1.0-8   SQUAREM_2017.10-1 crayon_1.3.4     
[55] R.oo_1.22.0      </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
