---
title: "Use FASH to analyze the stem cell gene expression data"
author: "Mengyin Lu"
date: "2018-08-07"
output: workflowr::wflow_html
---

## Data description
We have gene expression (microarray) data from [Burrows et al (2016), "Genetic Variation, Not Cell Type of Origin, Underlies the Majority of Identifiable Regulatory Differences in iPSCs"](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005793).

There are four individuals - Ind1, Ind2, Ind3, Ind4. Each individual has four samples types - Fibroblast, LCL, F-iPSC, L-iPSC. There are three replicates of the L-iPSCs A-C.

L-iPSC referes to iPSCs derived from LCLs, F-iPSC refers to iPSCs derived from Fibroblasts.

They were interested in the proportion of variation explained by cell type of origin vs the proportion of variation explained by individual in the iPSCs. They performed a linear mixed model with a fixed effect for cell type of origin (ie L-iPSC vs F-iPSC) and a random effect for individual. This model did no use the LCLs or the Fibroblasts from these individuals. 

## Data analysis
We use the naive F-test (ANOVA) and FASH to analyze the proportion of variance explained by cell type or individual. Specifically, we assume that gene expression $y_{gij}$ comes from the following model:
\[y_{gij} = \mu_g+\beta_{g,c(i)}+\gamma_{g,d(j)}+e_{gij},\]
where $g,i,j$ are the indices for gene, individual and cell type respectively. $\beta$ and $\gamma$ are random effects for individuals and celltypes respectively. Suppose $\beta_{g,c(i)}\sim N(0,v_{g}^{(ind)}$, $\gamma_{g,c(i)}\sim N(0,v_{g}^(ct)$ and $e_{gij}\sim N(0, v_{g}^{(err)})$, we are interested in estimating the "PVE"s defined as follows:
\[PVE_g^{(ind)}:=\frac{v_{g}^{(ind)}}{v_{g}^{(ind)}+v_{g}^{(ct)}+v_{g}^{(err)}},\]
\[PVE_g^{(ct)}:=\frac{v_{g}^{(ct)}}{v_{g}^{(ind)}+v_{g}^{(ct)}+v_{g}^{(err)}}.\]

Below are the R functions for our data analysis:
```{r}
library(limma)
library(qvalue)
library(ashr)

# naive F test for a single gene
get_fstat = function(y,condition1,condition2){
  fit = lm(y~condition1+condition2)
  fstat1 = anova(fit)$F[1]
  fstat2 = anova(fit)$F[2]
  MST1 = anova(fit)$M[1]
  MST2 = anova(fit)$M[2]
  MSE = anova(fit)$M[3]
  pval1 = unlist(anova(fit)["Pr(>F)"])[1]
  pval2 = unlist(anova(fit)["Pr(>F)"])[2]
  res = c(fstat1,fstat2,MST1,MST2,MSE,pval1,pval2)
  names(res) = c("fstat1","fstat2","MST1","MST2","MSE","pval1","pval2")
  return(res)
}

# Variance ratio = var_trt/var_err
# Variance ratio from the raw MSTs & MSEs
vr_ftest = function(nsamp,MST,MSE){
  sigma.c2.hat = pmax(0,(MST-MSE)/nsamp)
  sigma.e2.hat = MSE
  return(sigma.c2.hat/sigma.e2.hat)
}

# Variance ratio = var_trt/var_err
# logalpha = (1+nsamp*var_trt/var_err)
# So: VR = (exp(logalpha)-1)/nsamp
vr_plugin = function(nsamp,fashobj){
  (exp(fashobj$result$PosteriorMean)-1)/nsamp
}

# main function for computing the PVEs
pve_analysis = function(data){
  celltype1 = factor(rep(c("F","L"), each=4))
  individual1 = factor(rep(1:4, 2))
  design1 = model.matrix(~celltype1+individual1)
  
  # naive F tests
  ftests = apply(data,1,get_fstat,condition1=celltype1,condition2=individual1)
  vr.ct.ftest = vr_ftest(4, ftests["MST1",],ftests["MSE",])
  vr.ind.ftest = vr_ftest(2, ftests["MST2",],ftests["MSE",])
  
  # fash
  lim = lmFit(data, design=design1)
  lim.ct = eBayes(lim[,2])
  Fstats.ct = lim.ct$F # shrunk F stats
  Fpval.ct = lim.ct$F.p.value
  fash.ct = ash(log(Fstats.ct),1,
                lik=lik_logF(df1=1,df2=lim.ct$df.total[1]),mixcompdist="+uniform")
  vr.ct.fash = vr_plugin(4, fash.ct)

  lim.ind = eBayes(lim[,3:5])
  Fstats.ind = lim.ind$F # shrunk F stats
  Fpval.ind = lim.ind$F.p.value
  fash.ind = ash(log(Fstats.ind),1,
                 lik=lik_logF(df1=3,df2=lim.ind$df.total[1]),mixcompdist="+uniform")
  vr.ind.fash = vr_plugin(2, fash.ind)
  
  # compute PVEs from variance ratios
  pve.ct.fash = vr.ct.fash/(vr.ct.fash+vr.ind.fash+1)
  pve.ct.ftest = vr.ct.ftest/(vr.ct.ftest+vr.ind.ftest+1)
  pve.ind.fash = vr.ind.fash/(vr.ct.fash+vr.ind.fash+1)
  pve.ind.ftest = vr.ind.ftest/(vr.ct.ftest+vr.ind.ftest+1)
  
  # boxplot
  # boxplot(pve.ct.fash, pve.ct.ftest, pve.ind.fash, pve.ind.ftest,
  #       names=c("ct.fash","ct.ftest","ind.fash","ind.ftest"),
  #       main="PVE of celltypes (ct) and individuals (ind)")

  return(list(pve.ind.ftest=pve.ind.ftest, pve.ind.fash=pve.ind.fash,
              pve.ct.ftest=pve.ct.ftest, pve.ct.fash=pve.ct.fash))
}
```

Note that this dataset has unbalanced design (three L-iPSC replicates but just one F-iPSC sample for each individual). However it is not straightforward to use FASH on unbalanced dataset for PVE analysis. Hence we choose an ad-hoc way: each time we simply use one of the three L-iPSC replicates to form a balanced dataset, and compare the results of three trials. Fortunately the three trials give very similar results. 

```{r,message=FALSE,warning=FALSE,cache=TRUE}
# load data
geneexp = read.delim("data/OriginGeneExpression_Normalized.txt")

# just use L-iPSC replicate A
geneexp.A = geneexp[,c("Ind1.F.iPSC","Ind2.F.iPSC","Ind3.F.iPSC","Ind4.F.iPSC",
                     "Ind1.L.iPSC.A","Ind2.L.iPSC.A","Ind3.L.iPSC.A","Ind4.L.iPSC.A")]

# just use L-iPSC replicate B
geneexp.B = geneexp[,c("Ind1.F.iPSC","Ind2.F.iPSC","Ind3.F.iPSC","Ind4.F.iPSC",
                     "Ind1.L.iPSC.B","Ind2.L.iPSC.B","Ind3.L.iPSC.B","Ind4.L.iPSC.B")]

# just use L-iPSC replicate C
geneexp.C = geneexp[,c("Ind1.F.iPSC","Ind2.F.iPSC","Ind3.F.iPSC","Ind4.F.iPSC",
                     "Ind1.L.iPSC.C","Ind2.L.iPSC.C","Ind3.L.iPSC.C","Ind4.L.iPSC.C")]

A = pve_analysis(geneexp.A) 
B = pve_analysis(geneexp.B) 
C = pve_analysis(geneexp.C) 

# show distribution of pve estimates by fash or F-test
boxplot(A$pve.ct.fash, A$pve.ct.ftest, A$pve.ind.fash, A$pve.ind.ftest,
        names=c("CT(fash)","CT(F-test)","IND(fash)","IND(F-test)"), 
        ylab="Estimated PVE",
        main="PVE estimates (with L-iPSC A)")
boxplot(B$pve.ct.fash, B$pve.ct.ftest, B$pve.ind.fash, B$pve.ind.ftest,
        names=c("CT(fash)","CT(F-test)","IND(fash)","IND(F-test)"), 
        ylab="Estimated PVE",
        main="PVE estimates (with L-iPSC B)")
boxplot(C$pve.ct.fash, B$pve.ct.ftest, C$pve.ind.fash, C$pve.ind.ftest,
        names=c("CT(fash)","CT(F-test)","IND(fash)","IND(F-test)"), 
        ylab="Estimated PVE",
        main="PVE estimates (with L-iPSC C)")
```

Burrows et al (2016) simply uses the limma DE pipeline and claims that only one gene is classified as DE between L-iPSC and F-iPSC. They also use simple ANOVA $R^2$ to record the variance explained by celltypes or individuals (note that $R^2$ is different from our defined "PVE" here) and conclude that "individual genetic background captures a much larger proportion of gene regulatory variation than cell type of origin". 

We see that the FASH results are generally consistent with their results, that almost all genes have almost zero celltype PVEs. Compared to the raw F-test PVE estimates (which are pretty noisy), I think FASH gives much more intuitive results. 

According to FASH results, both datasets with L-iPSC replicate A or C suggest that gene "ENSG00000198825" is differentially expressed between L-iPSC and F-iPSC (FDR=0.05), but the dataset with L-iPSC replicate B does not any give significant DE genes (FDR=0.05). We check this gene:
```{r}
geneexp[which.max(A$pve.ct.fash),]
```

For this gene, individual 2 L-iPSC replicate B's gene expression level is much higher (>1.4) than that of replicates A and C. Compared to the typical gene expression variation within L-iPSC replicates, this seems somewhat suspicious...?
```{r}
hist(apply(geneexp[,c("Ind2.L.iPSC.A","Ind2.L.iPSC.B","Ind2.L.iPSC.C")],1,sd), 10,
     xlab="sd of replicates A-C for ind2",
     main="Histogram of sd")
```

## Session Information

```{r session-info}
```

